beacon-chain/sync/custody.go

```diff
@@ -103,12 +103,35 @@ func (s *Service) custodyGroupCount(context.Context) (uint64, error) {
 		return cfg.NumberOfCustodyGroups, nil
 	}
 
+	// Calculate validator custody requirements
 	validatorsCustodyRequirement, err := s.validatorsCustodyRequirement()
 	if err != nil {
 		return 0, errors.Wrap(err, "validators custody requirement")
 	}
 
-	return max(cfg.CustodyRequirement, validatorsCustodyRequirement), nil
+	effectiveCustodyRequirement := max(cfg.CustodyRequirement, validatorsCustodyRequirement)
+
+	// If we're not in semi-supernode mode, just use the effective requirement.
+	if !flags.Get().SemiSupernode {
+		return effectiveCustodyRequirement, nil
+	}
+
+	// Semi-supernode mode custodies at least half of all custody groups (e.g. 64 out of 128),
+	// which is the minimum required for blob reconstruction.
+	semiSupernodeTarget := cfg.NumberOfCustodyGroups / 2
+	if effectiveCustodyRequirement <= semiSupernodeTarget {
+		return semiSupernodeTarget, nil
+	}
+
+	// If effective requirement is higher than semi-supernode target, use it.
+	log.WithFields(logrus.Fields{
+		"semiSupernodeTarget":          semiSupernodeTarget,
+		"effectiveCustodyRequirement":  effectiveCustodyRequirement,
+		"custodyRequirement":           cfg.CustodyRequirement,
+		"validatorsCustodyRequirement": validatorsCustodyRequirement,
+	}).Debug("Custody requirements exceed semi-supernode mode; using higher custody count to meet requirements.")
```

Comment:
Do we need to print that every minute?

Reply:
probably not i guess I should remove this